lib/malloc.cpp

```diff
@@ -363,26 +373,35 @@ namespace quda {
     }
     if (alloc[HOST].count(ptr)) {
       track_free(HOST, ptr);
+      free(ptr);
     } else if (alloc[PINNED].count(ptr)) {
       cudaError_t err = cudaHostUnregister(ptr);
       if (err != cudaSuccess) {
 	printfQuda("ERROR: Failed to unregister pinned memory (%s:%d in %s())\n", file, line, func);
 	errorQuda("Aborting");
       }
       track_free(PINNED, ptr);
+      free(ptr);
     } else if (alloc[MAPPED].count(ptr)) {
+#ifdef HOST_ALLOC
+      cudaError_t err = cudaFreeHost(ptr);
+      if (err != cudaSuccess) {
+	printfQuda("ERROR: Failed to free host memory (%s:%d in %s())\n", file, line, func);
+	errorQuda("Aborting");
+      }
+#else
       cudaError_t err = cudaHostUnregister(ptr);
```

Comment:
do we miss a free(ptr) here if HOST_ALLOC is not defined?

Reply:
Good catch.  Will push a fix